pkg/pravega/pravega_segmentstore.go

```diff
@@ -34,6 +35,21 @@ const (
 )
 
 func deploySegmentStore(pravegaCluster *api.PravegaCluster) (err error) {
+	err = sdk.Create(makeSegmentStoreHeadlessService(pravegaCluster))
```

Comment:
Is this necessary even in the case that external access is disabled?

Reply:
Please bear with me, I'm almost there. A few points that called my attention:

- This call to make each segment store a headless service wasn't there before, yet we were able to get Pravega up and running. If that's the case, then d we still need to make this call in the absence of external access?
- I'm not sure I understand when you say that segment store instances communicate with other. I think you mean to say controller to segment store communication, as segment store instances do not talk to each other as of today.

Reply:
> This call to make each segment store a headless service wasn't there before, yet we were able to get Pravega up and running. If that's the case, then do we still need to make this call in the absence of external access?

You are right. This is actually not necessary for external connectivity to work, but it was added there as a best practice for internal communication when creating stateful apps in K8.

> I'm not sure I understand when you say that segment store instances communicate with each other. I think you mean to say controller to segment store communication, as segment store instances do not talk to each other as of today.

Correct. Controller communication with Segmentstores happens via external IP address when external connectivity is enabled. Even though this works, it's not ideal as we are adding unnecessary hops to the traffic route. Ideally, the Controller would use the headless service to communicate with individual Segmentstores. Before we can get this implemented in the operator, Pravega needs to support different endpoints for inter-component communication and external clients.